zwavegeneric/integrationpluginzwavegeneric.cpp

```diff
@@ -0,0 +1,203 @@
+            if (value.genre() == ZWaveValue::GenreUser && value.commandClass() == ZWaveValue::CommandClassDoorLock && value.index() == 0) {
+                thing->setStateValue(doorLockStateStateTypeId, value.value().toBool() ? "locked" : "unlocked");
+            }
+        });
+    }
+
+    info->finish(Thing::ThingErrorNoError);
+}
+
+void IntegrationPluginZWaveGeneric::executeAction(ThingActionInfo *info)
+{
+    if (!hardwareManager()->zwaveResource()->available()) {
+        info->finish(Thing::ThingErrorHardwareNotAvailable);
+        return;
+    }
+
+    Thing *thing = info->thing();
+    ZWaveNode *node = nodeForThing(info->thing());
+    if (!node->reachable()) {
+        info->finish(Thing::ThingErrorHardwareNotAvailable);
+        return;
+    }
+
+    if (thing->thingClassId() == powerSocketThingClassId) {
+        if (info->action().actionTypeId() == powerSocketPowerActionTypeId) {
+            ZWaveValue powerValue = node->value(ZWaveValue::GenreUser, ZWaveValue::CommandClassSwitchBinary, 1, 0, ZWaveValue::TypeBool);
+            powerValue.setValue(info->action().paramValue(powerSocketPowerActionPowerParamTypeId).toBool());
+            node->setValue(powerValue);
+            thing->setStateValue(powerSocketPowerStateTypeId, info->action().paramValue(powerSocketPowerActionPowerParamTypeId));
+            info->finish(Thing::ThingErrorNoError);
+            return;
+        }
+    }
+
+    if (thing->thingClassId() == powerMeterSocketThingClassId) {
+        if (info->action().actionTypeId() == powerMeterSocketPowerActionTypeId) {
+            ZWaveValue powerValue = node->value(ZWaveValue::GenreUser, ZWaveValue::CommandClassSwitchBinary, 1, 0, ZWaveValue::TypeBool);
+            powerValue.setValue(info->action().paramValue(powerMeterSocketPowerActionPowerParamTypeId).toBool());
+            node->setValue(powerValue);
+            thing->setStateValue(powerMeterSocketPowerStateTypeId, info->action().paramValue(powerMeterSocketPowerActionPowerParamTypeId));
+            info->finish(Thing::ThingErrorNoError);
+            return;
+        }
+    }
+
+    if (thing->thingClassId() == doorLockThingClassId) {
+        if (info->action().actionTypeId() == doorLockLockActionTypeId) {
+            ZWaveValue lockValue = node->value(ZWaveValue::GenreUser, ZWaveValue::CommandClassDoorLock, 1, 0, ZWaveValue::TypeBool);
+            lockValue.setValue(false);
+            node->setValue(lockValue);
+//            thing->setStateValue(doorLockStateStateTypeId, "locked");
+            info->finish(Thing::ThingErrorNoError);
+            return;
+        }
+        if (info->action().actionTypeId() == doorLockUnlockActionTypeId) {
+            ZWaveValue lockValue = node->value(ZWaveValue::GenreUser, ZWaveValue::CommandClassDoorLock, 1, 0, ZWaveValue::TypeBool);
+            lockValue.setValue(true);
+            node->setValue(lockValue);
+//            thing->setStateValue(doorLockStateStateTypeId, "unlocked");
```

Comment:
not updating the state?

Reply:
hmm... the device will send a value changed which in turns updates the state again... Given I don't have such a device, it's a bit hard to tell how it would behave better...